src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkTask.java

```diff
@@ -155,11 +160,16 @@ protected String convertTopicToIndexName(final String topic) {
     }
 
 
-    private void ensureIndexExists(final String index) {
+    private void ensureIndexOrDataStreamExists(final String index) {
         if (!indexCache.contains(index)) {
             if (!client.indexExists(index)) {
```

Comment:
I think this logic is flawed now: what if `client.indexExists(index))` return `true` but index is not data stream?

Reply:
no it is not, since `openSearchClient::indexExists` checks existing data streams (was added before). The name of the method is misleading and needs to be changed to `resourceExists` or `indexOrDataStreamExists`. 
So in case you already have a pre-existing data stream the connector will use it as is the only difference in this case that it starts to use CREATE operation type which is faster compare to INSERT.

Reply:
So I think I will change the name from  `indexExists` to `indexOrDataStreamExists` WDYT?

Reply:
`indexOrDataStreamExists` makes sense, thank you